### PR TITLE
Generating metas

### PR DESCRIPTION
**Pull request by littlejohntj**

### CODE REVIEW

File: playground/src/main/java/com/solanamobile/buoy/playground/Main.kt [hunk 1]
```diff
@@ -32,40 +74,62 @@ fun main(arguments: Array<String>) {
...
+        // the pubkey and generate the AccountMeta item in the list
+        instruction.args.forEach { arg ->
+            val type = classTypeForArgType(arg.type)
+            functionBuilder
+                .addParameter(arg.name, type)
+        }
+
         instruction.accounts.forEach { metaDescriptor ->
+
+            val metaItem = String.format("\tAccountMeta(publicKey = %s, isSigner = %s, isWritable = %s)", metaDescriptor.name, metaDescriptor.isSigner.toString(), metaDescriptor.isMut.toString())
```
Comment (creativedrewy) [inline playground/src/main/java/com/solanamobile/buoy/playground/Main.kt:93]: Just want to make sure that AccountMeta is imported properly into the resulting class

**Review by creativedrewy — APPROVED** [review]
Really good progress!

File: playground/src/main/java/com/solanamobile/buoy/playground/Main.kt [hunk 2]
```diff
@@ -32,40 +74,62 @@ fun main(arguments: Array<String>) {
...
+            functionBuilder
+                .addStatement(String.format("val %sBytes = %s.toBytes()", arg.name, arg.name))
+            instructionDataStatementBuilder.append( String.format(" + %sBytes", arg.name, arg.name) )
+        }
+
+        val instructionDataStatement = instructionDataStatementBuilder.toString()
+
+        functionBuilder.addStatement(instructionDataStatement)
+
+        functionBuilder.addStatement("return TransactionInstruction(")
```
[the remark this answers is not in the document]
Comment (littlejohntj) [reply playground/src/main/java/com/solanamobile/buoy/playground/Main.kt:122]: agreed! honestly I think I just wanted core functionality to start shaping and I'll hit it w/ a refactor soon #famouslastwords

File: playground/src/main/java/com/solanamobile/buoy/playground/Main.kt [hunk 3]
```diff
@@ -24,6 +65,7 @@ fun main(arguments: Array<String>) {
 
     //Use the objects to generate the contract file
     val contractFile = FileSpec.builder("com.solanamobile.buoy", idlSource.name)
```
[the remark this answers is not in the document]
Comment (littlejohntj) [reply playground/src/main/java/com/solanamobile/buoy/playground/Main.kt:67]: yes agree'd

File: playground/src/main/java/com/solanamobile/buoy/playground/Main.kt (hunk 1 above)
Comment (littlejohntj) [reply playground/src/main/java/com/solanamobile/buoy/playground/Main.kt:93]: I'm not sure how to do that lol but I will likely learn soon

**Pull request merged**